src/node/history.h

```diff
@@ -745,7 +757,8 @@ namespace ccf
             root,
             hashed_nonce,
             primary_sig,
-            replicated_state_tree.serialise());
+            replicated_state_tree.serialise(
+              commit_txid.second, txid.version - 1));
```

Comment:
This isn't quite right, it suffers from the same problem as now: the commit index may be quite a bit behind the last signature.

Reply:
I was wondering about that too. How do I find the index of the previous signature?

Reply:
See below; is https://github.com/microsoft/CCF/pull/2053/commits/c53485564b1f67c08ebabe7efdf4cfbb7fabd19c the right way to get that index?

Reply:
All done! I think I have the right indices now and the CI passed all tests.

Reply:
Thank you, looks good!